pkg/kubelet/kubelet_pods.go

```diff
@@ -414,6 +414,44 @@ func (kl *Kubelet) makeEnvironmentVariables(pod *v1.Pod, container *v1.Container
 		return result, err
 	}
 
+	var (
+		configMaps = make(map[string]*v1.ConfigMap)
+		tmpEnv     = make(map[string]string)
+	)
+	for _, envFrom := range container.EnvFrom {
+		if envFrom.ConfigMapRef != nil {
+			name := envFrom.ConfigMapRef.Name
+			configMap, ok := configMaps[name]
+			if !ok {
+				if kl.kubeClient == nil {
```

Comment:
This type of check happens everywhere because there is no guarantee at the time of creation. It seems to allow a nil kubeClient.